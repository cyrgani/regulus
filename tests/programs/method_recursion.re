type(I, x,
    =(half, fn(self, _(
        =(x, .(self, x)),
        if(
            >(x, 0),
            _(
                print("halving", x),
                =(s2, I(/(x, 2))),
                @(s2, decr),
            )
        )
    ))),
    =(decr, fn(self, _(
        =(x, .(self, x)),
        if(
            >(x, 0),
            _(
                print("decrementing", x),
                =(s2, I(-(x, 1))),
                @(s2, half),
            )
        )
    )))
),

=(i, I(10)),
@(i, decr),
