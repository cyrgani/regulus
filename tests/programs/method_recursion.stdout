decrementing 10 
halving 9 
decrementing 4 
halving 3 
decrementing 1 
